var/spack/repos/builtin/packages/f18/package.py

```diff
@@ -30,6 +30,7 @@ class F18(CMakePackage):
 
     # Conflicts
     compiler_warning = 'F18 requires a compiler with support for C++17'
+    conflicts('%apple-clang@:6', msg=compiler_warning)
```

Comment:
Apple Clang versions don't necessarily match Clang versions

Reply:
I'll fix this. In the meantime posting a wiki with the conversion list so others will have access to it.

https://en.wikipedia.org/wiki/Xcode#Latest_versions

Reply:
Fixed